server/src/main/java/com/epam/aidial/core/server/vertx/stream/BlobWriteStream.java

```
@@ -179,16 +177,14 @@ public WriteStream<Buffer> drainHandler(Handler<Void> handler) {
             synchronized (BlobWriteStream.this) {
                 try {
                     if (mpu == null) {
-                        mpu = storage.initMultipartUpload(resource.getAbsoluteFilePath(), contentType);
-                        etagBuilder = new EtagBuilder();
+                        mpu = resourceService.initFileUpload(resource, contentType, etag, author);
```

Please try to remove etag validation in UploadController.

---

we do etag validation in the controller at very begining because we want to fail fast request before we starting upload the first chunk
 
I suggest to remove etag validation in initFileUpload instead of the controller